src/nbsphinx.py

```
@@ -1886,10 +1886,13 @@ def patched_toctree_resolve(self, docname, builder, toctree, *args, **kwargs):
 
 
 def config_inited(app, config):
-    if '.ipynb' not in config.source_suffix:
-        app.add_source_suffix('.ipynb', 'jupyter_notebook')
-    for suffix in config.nbsphinx_custom_formats:
-        app.add_source_suffix(suffix, 'jupyter_notebook')
+    suffixes = [".ipynb", *config.nbsphinx_custom_formats]
```

I don't think that the `nbsphinx_custom_formats` have to be checked further. They are provided by the user, so it's the user's responsibility not to provide conflicting suffixes.

---

I see. Does this `nbsphinx_custom_formats` config do anything else that sphinx's `source_suffix` don't do? https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-source_suffix

> so it's the user's responsibility not to provide conflicting suffixes

Actually my understanding is that sphinx's `source_suffix` is a way for users to overwrite anything that extensions have setup, so it's not really the same thing? Or am I completely misunderstanding all this? (sphinx beginner here)

---

> Does this `nbsphinx_custom_formats` config do anything else that sphinx's `source_suffix` don't do?

Yes. In addition to registering the source suffix, it also provides a function that transforms a string containing certain custom format into a Jupyter notebook object. See https://nbsphinx.readthedocs.io/custom-formats.html.

But I assume that if a user provides a custom format, they don't want their chosen suffix to be overwritten by some other Sphinx extension.

And if another extension is loaded after `nbsphinx`, I prefer an error over silently ignoring the source suffix.

And if a user provides `nbsphinx_custom_formats` *and* a conflicting `source_suffix`, it's their own fault.

---

@martinRenou Do you want to update this PR or should I do it?

---

Please do it if you can :) Thanks a lot!

---

I have rebased your branch and I've added fc79fd1ac5ab6df225b256ef4ca203c58c5ad92b that unconditionally adds `nbsphinx_custom_formats`.

Can you please test if that works for you?

---

Thanks a lot!

---

I've merged this PR already, but please let me know if it doesn't work as expected, so we can fix that.